pkg/archive/archive.go

```diff
@@ -37,11 +37,13 @@ type (
 		Compression      Compression
 		NoLchown         bool
 		ChownOpts        *TarChownOptions
-		Name             string
```

Comment:
Rationale for removing this: It is never used, we have logic that uses it below, but I replaced it with the `rebaseNames` map.